### ISSUE TITLE
Cumbersome to edit time [regression]

### ISSUE THREAD

**Issue by CendioOssman** [projecthamster/hamster#439]
It seems the 2.x code lost the separate input widgets for start and end time. This makes it rather annoying to modify time as the previous interface had quick suggestions for common intervals.

It looks like this was removed in 697a03e59312c72135788ba77ce09bd0ac0c6e0d. But I don't see any rationale. Did someone consider the feature pointless? Or was it removed because no one had ported the supporting code?

Would a PR be accepted returning the old behaviour?

**Comment by CendioOssman** [comment]
How it looks in 1.04:

![2019-09-03-152142_3840x1200_scrot](https://user-images.githubusercontent.com/6640916/64176972-cb071f00-ce5e-11e9-999c-6406d9c7d13a.png)

**Comment by GeraldJansen** [comment]
Just a guess, but I think the new approach of specifying times on the activity input line was done to harmonize input in the hamster edit windows and the gnome hamster-shell-extension. 

Personally I really like the new method, because it is very keyboard friendly. With the current master branch (i.e. later than v2.2.2), the start-end times are highlighted when you open the update window. A single keystroke moves you to the start or end and a space deletes the time(s). New times can be entered as 4 digits without the ':'.  Together with other keyboard shortcuts (#230, #387) and a window manager shortcut to open the hamster overview window, I now mainly use hamster through the keyboard.

**Comment by CendioOssman** [comment]
Hmm... Alright. I guess I'll have to test the new code and see how it feels.

For me I saw two fairly different use cases here, that the old UI catered to:

 - Starting a new activity, where a single input field with some syntax shortcuts was/is fantastic
 - Editing/correcting an existing entry where it was more important to quickly adjust just a single detail without having to risk other information (which hopefully were already correct)

**Comment by WBTMagnum** [comment]
I'm still irritated by the single input field. I really like it in the hamster-shell-extension, where I can quickly start and enter tasks. I also see the point on the CLI. For editing past and running tasks in the panel I would prefer separate input fields.

Just my 2 cents

**Comment by ederag** [comment]
@CendioOssman and @WBTMagnum, 
thanks for explaining well why separate fields for *editing* are important,
all the more if you are using overview mainly for editing.
As @GeraldJansen nicely expressed, the command-line style entry has some advantages though.

One idea would be to separate "Add" and "Edit", not satisfying because cloning is in-between.
Maybe we could combine them ?
Got some ideas, need some time before discussing them.

Not forgetting the original issue; would it be alleviated by point 3 of [this proposition](https://github.com/projecthamster/hamster/wiki/Entering-activities/873c6479944eb680fc7295054a7574bc158d1e52#time) ?
Or is the goal to be able to enter the end just with mouse ?

Bottom line: understood, not easy, lots of hard changes already being worked on, please be patient.

**Comment by CendioOssman** [comment]
> Not forgetting the original issue; would it be alleviated by point 3 of [this proposition](https://github.com/projecthamster/hamster/wiki/Entering-activities/873c6479944eb680fc7295054a7574bc158d1e52#time) ?
> Or is the goal to be able to enter the end just with mouse ?
> 

Being able to use the mouse only would be preferred. The reason being that it's always annoying to have to switch. And you have to use the mouse to select the entry you want to modify and then click to edit. So being able to finish your task without switching to the keyboard would be very nice. That would also include things like clicking on the activity text and getting good suggestions for the scenario where you've entered the wrong one.

> Bottom line: understood, not easy, lots of hard changes already being worked on, please be patient.

Of course. I'm only filing this issue to make sure my use case isn't completely overlooked. :)

**Comment by ederag** [comment]
> always annoying to have to switch

That's exactly why it is now possible to do everything through keyboard, 
including selection of the entry to be modified.
Tab traversal improvements are planned.

But again, I understand the need for well isolated edition of fields.
Suggestions to allow mouse-only edition would be nice then, indeed.

**Comment by GeraldJansen** [comment]
> And you have to use the mouse to select the entry you want to modify and then click to edit.

Just to clarify @ederag's response that this is possible through the keyboard, for others that might be interested, just use the Up or Down keys to move through the list (Up first to start from bottom and Down first to start from top), move to the row of interest then press Enter to open the Edit Activity dialog. Admittedly this is not obvious and documentation is needed.

**Comment by ederag** [comment]
Good news. `TimeInput` is still *almost* functional (not pushed yet). Should be there in v3.0.

**Comment by ederag** [comment]
Hopefully fixed with PR #461.
Please provide feedback there.

**Comment by ederag** [comment]
@CendioOssman That's fully understandable, thanks for letting us know.
@WBTMagnum, @Forage Your feedback on design decisions in #461 would be welcome as well.

**Comment by ederag** [comment]
@CendioOssman PR #461 is far from perfect, but seems to have converged.
It could be merged this week-end, but given your past contributions, especially on completion,
it looks just fair to give you another opportunity to comment; would you like another week ?

**Comment by CendioOssman** [comment]
Sorry for the delay. The new interface looks really nice. Fantastic work. :)

This should satisfy my needs with regard to editing time. From what I can tell I can keep my workflow from 1.04 without any substantial change.

**Comment by ederag** [comment]
Closed by #461.
Thanks for the report !
